src/js/camresolutionstest.js

```diff
@@ -36,148 +20,109 @@ var resolutions = [[160, 120, false],
+});
 
-function cameraTest_(resolutions) {
-  var test = new CamResolutionsTest(resolutions);
-  test.run();
-}
+addTest(testSuiteName.CAMERA, testCaseName.CHECKRESOLUTION480, function(test) {
+  var camResolutionsTest = new CamResolutionsTest(test);
+  camResolutionsTest.run([640, 480]);
+});
+
+addTest(testSuiteName.CAMERA, testCaseName.CHECKRESOLUTION720, function(test) {
+  var camResolutionsTest = new CamResolutionsTest(test);
+  camResolutionsTest.run([1280, 720]);
+});
 
-function CamResolutionsTest(resolutionArray) {
-  this.resolutions = resolutionArray;
+function CamResolutionsTest(test) {
+  this.test = test;
   this.mandatoryUnsupportedResolutions = 0;
-  this.numResolutions = this.resolutions.length;
   this.counter = 0;
   this.supportedResolutions = 0;
   this.unsupportedResolutions = 0;
   this.currentResolutionForCheckEncodeTime = null;
-
   this.isMuted = false;
-  this.stream = null;
-}
-
-function resolutionMatchesIndependentOfRotationOrCrop_(aWidth, aHeight,
-                                                       bWidth, bHeight) {
-  var minRes = Math.min(bWidth, bHeight);
-  return (aWidth === bWidth && aHeight === bHeight) ||
-         (aWidth === bHeight && aHeight === bWidth) ||
-         (aWidth === minRes && bHeight === minRes);
 }
 
 CamResolutionsTest.prototype = {
-  run: function() { this.triggerGetUserMedia_(this.resolutions[0]); },
-
-  triggerGetUserMedia_: function(resolution) {
+  run: function(resolution) {
     var constraints = {
       audio: false,
-      video: {
-        mandatory: {
-          minWidth:  resolution[0],
-          minHeight: resolution[1],
-          maxWidth:  resolution[0],
-          maxHeight: resolution[1]
-        }
-      }
+      video: {width: {exact: resolution[0]}, height: {exact: resolution[1]}}
     };
-    try {
-      doGetUserMedia(constraints, this.successFunc_.bind(this),
-          this.failFunc_.bind(this));
-    } catch (e) {
-      reportFatal('GetUserMedia failed.');
-    }
-  },
-
-  successFunc_: function(stream) {
-    this.stream = stream;
-    this.supportedResolutions++;
-    var selectedResolution = this.resolutions[this.counter++];
-    // Measure performance only when testing one resolution.
-    if (selectedResolution[2] && this.numResolutions === 1) {
-      this.collectAndAnalyzeStats_(stream, selectedResolution);
-    } else {
-      reportInfo('Supported ' + selectedResolution[0] + 'x' +
-                 selectedResolution[1]);
-      stream.getVideoTracks()[0].stop();
-      this.finishTestOrRetrigger_();
-    }
+    navigator.mediaDevices.getUserMedia(constraints)
+    .then(function(stream) {
+      this.collectAndAnalyzeStats_(stream, resolution);
+    }.bind(this))
+    .catch(function(error) {
+      this.test.reportError('getUserMedia failed with error: ' + error);
+      this.test.done();
+    }.bind(this));
   },
 
-  collectAndAnalyzeStats_: function(stream, selectedResolution) {
+  collectAndAnalyzeStats_: function(stream, resolution) {
```

Comment:
If this test only verifies one resolution then fetch that value from this.resolution